### PR TITLE
fix: android build issue on clean repository

### LINKED ISSUE #120: bug: EAS Building an apk from a clean repo fails
### Provide environment information

```
System:
  OS: Windows 10 10.0.19045
  CPU: (24) x64 AMD Ryzen 9 3900X 12-Core Processor
  Memory: 21.68 GB / 31.95 GB
Binaries:
  Node: 16.15.1 - C:\Program Files\nodejs\node.EXE
  npm: 8.14.0 - ~\AppData\Roaming\npm\npm.CMD
```

### Describe the bug

Building an apk using preview profile fails on gradlew on eas.

Originally fixed this unreliably by prebuilding locally and adding `reactNativeVersion = "0.70.5"` to apps/expo/android/build.gradle

expo doctor gives no errors

Continued development with the issue returning and now can't fix it, even with a clean create-t3-turbo repo.

Building clean repo (without prebuilding locally) gives this error on Run gradlew:
```bash
[stderr] FAILURE: Build completed with 2 failures.
[stderr] 1: Task failed with an exception.
[stderr] -----------
[stderr] * Where:
[stderr] Build file '/home/expo/workingdir/build/node_modules/expo/android/build.gradle' line: 9
[stderr] * What went wrong:
[stderr] A problem occurred evaluating project ':expo'.
[stderr] > Cannot invoke method multiply() on null object
[stderr] * Try:
[stderr] > Run with --stacktrace option to get the stack trace.
[stderr] > Run with --info or --debug option to get more log output.
[stderr] > Run with --scan to get full insights.
[stderr] ==============================================================================
[stderr] 2: Task failed with an exception.
[stderr] -----------
[stderr] * Where:
[stderr] Build file '/home/expo/workingdir/build/node_modules/expo/android/build.gradle' line: 78
[stderr] * What went wrong:
[stderr] A problem occurred configuring project ':expo'.
[stderr] > Could not get unknown property 'release' for SoftwareComponentInternal set of type org.gradle.api.internal.component.DefaultSoftwareComponentContainer.
[stderr] * Try:
[stderr] > Run with --stacktrace option to get the stack trace.
[stderr] > Run with --info or --debug option to get more log output.
[stderr] > Run with --scan to get full insights.
[stderr] ==============================================================================
[stderr] * Get more help at https://help.gradle.org
[stderr] BUILD FAILED in 1m 49s
```

The prebuild stage skips metro:
```
› Metro skipped: Project metro.config.js does not match prebuild template.
```
and also has these warnings about peer dependencies:
```
WARN  Issues with peer dependencies found
apps/expo
└─┬ expo-splash-screen
  └─┬ @expo/prebuild-config
    └── ✕ missing peer expo-modules-autolinking@>=0.8.1
Peer dependencies that should be installed:
  expo-modules-autolinking@>=0.8.1
packages/auth
└─┬ @next-auth/prisma-adapter
  └── ✕ missing peer @prisma/client@">=2.26.0 || >=3"
Peer dependencies that should be installed:
  @prisma/client@">=2.26.0 || >=3"
Packages: +1291
```

Thanks for any help, will be happy to provide any more info, I didn't create a reproduction repo purely because I don't actually change anything (Except for in app.json but this will need to be for your own expo account/project).

I am on windows so I cannot use EAS locally, so I'm having to wait 30 minuites in a queue just for my build to fail, so trying to debug this is difficult.

I'd appreciate if somebody could try to reproduce, so I can determine whether it is my environment causing the issues? Maybe my pnpm installation?

### Link to reproduction

https://github.com/t3-oss/create-t3-turbo

### To reproduce

git clone https://github.com/t3-oss/create-t3-turbo
cd create-t3-turbo
pnpm i

// update lines in apps/expo/app.json
4 - "slug": "actual-slug-of-project",
22 - "bundleIdentifier": "ext.yourdomain.projectname"
35 - "projectId": "your-expo-project-id-xxxxxxxxxxxx"

pnpm add -g eas-cli
eas login // Login with account that owns project

cd apps/expo
eas build:configure # Select All

eas build --platform android --profile preview

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by PiotrekPKP**
# Fix for Android build issue
### Fixes #120 

This is a one-file fix, that's why I haven't used TypeScript but `//@ts-check` - creating entire ts workspace just for a simple expo fix would not be worth it I think.

## Changelog
- Added a plugin that appends react native and expo versions to `build.gradle` file
- Appending those versions fixes the build error described in issue #120
- also added splash screen package which expo really wanted for some reason (I can remove it if really needed)

### CODE REVIEW

**Comment by juliusmarminge** [comment]
> also added splash screen package which expo really wanted for some reason (I can remove it if really needed)

god I which they just bundled some stuff into the main `expo`... or at the very least ship their "templates" with all the peerDeps included

**Comment by juliusmarminge** [comment]
Is this some generic fix for monorepos or why is this needed? We're using all the correct react-native / react versions for the current Expo SDK...

**Comment by BRDSLY** [comment]
> Is this some generic fix for monorepos or why is this needed? We're using all the correct react-native / react versions for the current Expo SDK...

I think EAS just can't detect what expo and react native version is being used, not monorepo specific, I'm guesssing its related to [this](https://github.com/expo/expo/issues/18129#issuecomment-1223817877), however it now also needs the expo version explicitly specified too...

**Comment by PiotrekPKP** [comment]
> Is this some generic fix for monorepos or why is this needed? We're using all the correct react-native / react versions for the current Expo SDK...

Tbh I don't really know, but I saw people saying they're encountering this issue on clean expo app.
This solution was described there and I tested it and is seems working... Heh
Issue in expo repo for reference: https://github.com/expo/expo/issues/18129

File: apps/expo/expo-plugins/with-modify-gradle.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+//@ts-check
```
Comment (juliusmarminge) [inline apps/expo/expo-plugins/with-modify-gradle.js:1]: add a comment here linking to our issue as well as the original issue on expo for future reference please

Comment (PiotrekPKP) [reply apps/expo/expo-plugins/with-modify-gradle.js:1]: Done 👍

**Review by juliusmarminge — APPROVED** [review]

**Comment by juliusmarminge** [comment]
Sorry just migrated to `app.config.ts`, you mind resolving the conflict?

**Comment by PiotrekPKP** [comment]
> Sorry just migrated to `app.config.ts`, you mind resolving the conflict?

Should be good. Sorry for the messy commits, I'm rarely doing that kid of stuff ;)

**Pull request merged**